libyul/optimiser/Suite.cpp

```diff
@@ -121,6 +123,12 @@ void OptimiserSuite::run(
 	{
 		yulAssert(_meter, "");
 		ConstantOptimiser{*dialect, *_meter}(ast);
+		if (dialect->providesObjectAccess())
```

Comment:
Doesn't it also have to check that it's EVM?

Reply:
This is already inside ``if (EVMDialect const* dialect = dynamic_cast<EVMDialect const*>(&_dialect))``

Reply:
Ok